src/elements/content-sidebar/activity-feed/task-new/AssigneeList.js

```diff
@@ -0,0 +1,162 @@
+// @flow strict
+import * as React from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import type { InjectIntlProvidedProps } from 'react-intl';
+import uniqueId from 'lodash/uniqueId';
+import PlainButton from '../../../../components/plain-button';
+import ReadableTime from '../../../../components/time/ReadableTime';
+import messages from '../../../common/messages';
+import AvatarGroupAvatar from './AvatarGroupAvatar';
+import { TASK_NEW_APPROVED, TASK_NEW_REJECTED, TASK_NEW_COMPLETED, TASK_NEW_NOT_STARTED } from '../../../../constants';
+
+import './AssigneeList.scss';
+
+const MIN_ASSIGNEES_SHOWN = 3;
+const TASKS_PAGE_SIZE = 20; // service does not return the page size to the client at the moment
+
+type Props = {|
+    getAvatarUrl: GetAvatarUrlCallback,
+    minAssignees: number,
+    onExpand: Function,
+    users: TaskAssigneeCollection,
+|} & InjectIntlProvidedProps;
+
+type State = {
+    isCollapsed: boolean,
+};
+
+const statusMessages = {
+    [TASK_NEW_APPROVED]: messages.tasksFeedStatusApproved,
+    [TASK_NEW_REJECTED]: messages.tasksFeedStatusRejected,
+    [TASK_NEW_COMPLETED]: messages.tasksFeedStatusCompleted,
+    [TASK_NEW_NOT_STARTED]: null,
+};
+
+const Datestamp = ({ date }: { date: ISODate | Date }) => {
+    return <ReadableTime timestamp={new Date(date).getTime()} alwaysShowTime relativeThreshold={0} />;
+};
+
+const AvatarDetails = React.memo(({ user, status, completedAt, className }) => {
+    const statusMessage = statusMessages[status] || null;
+    return (
+        <div className={className}>
+            <div className="bcs-AssigneeList-detailsName">{user.name}</div>
+            {statusMessage && completedAt && (
+                <div className="bcs-AssigneeList-detailsStatus">
+                    <FormattedMessage {...statusMessage} values={{ dateTime: <Datestamp date={completedAt} /> }} />
+                </div>
+            )}
+        </div>
+    );
+});
+
+class AssigneeList extends React.Component<Props, State> {
+    state = {
+        isCollapsed: true,
+    };
+
+    static defaultProps = {
+        minAssignees: MIN_ASSIGNEES_SHOWN,
+        users: {},
+    };
+
+    listTitleId = uniqueId('assignee-list-title-');
+
+    showLessAssignees = () => {
+        this.setState({
+            isCollapsed: true,
+        });
+    };
+
+    showMoreAssignees = async () => {
```

Comment:
Would using a promise allow you to skip the empty catch case?

Reply:
If I use a promise it would look something like this? 

```
onExpand().then(setState).catch(setState);
```

not sure if there's a better way to skip the empty catch case w/o repeating the set state call